ui/index.d.ts

```diff
@@ -1616,3 +1616,9 @@ export declare abstract class AzExtTreeFileSystem<TItem extends AzExtTreeItem> i
  * Registers a command that will prompt users with a list of issues they can report from that session of VS Code
  */
 export function registerReportIssueCommand(commandId: string): void;
+
+/**
+ * Gets or generates a unique scheme so that multiple extensions using this same code don't conflict with each other
+ * Use to provide scheme for various TextDocument providers
+ */
+export function getOrGenerateContentScheme(): string;
```

Comment:
Can I see how you're using this? This isn't something I ever thought would be exported from this package. It's very much an internal implementation detail

Reply:
Yeah, so I'm registering it here (on activate): https://github.com/microsoft/vscode-azurestaticwebapps/blob/nat/actionLog/src/extension.ts#L36

For this FoldingProvider https://github.com/microsoft/vscode-azurestaticwebapps/blob/nat/actionLog/src/commands/github/GitHubLogFoldingProvider.ts

For any file opened here: https://github.com/microsoft/vscode-azurestaticwebapps/blob/nat/actionLog/src/commands/github/openActionLog.ts#L26

Reply:
I think the alternative would be to implement my own version of `openReadOnlyContent` within SWA

Reply:
I don't like re-implementing `openReadOnlyContent`

The `DocumentSelector` you pass to `registerFoldingRangeProvider` doesn't have to filter by scheme. What if it filters by file path and you add some unique string to the file path for github logs? That's theoretically better in the sense that the `FoldingProvider` shouldn't actually apply to _all_ readonly content, just the github logs. In practice I admit it probably doesn't matter since other readonly content (like "View Properties") probably won't have the regions anyways.

Let me know if that doesn't work for some reason. If we end up going with your current PR, I would like to "hide" some of the internal details (like the fact that we generate the scheme) and do something a bit more generic like this:
```
/**
 * Gets the scheme used for read only content
 */
export function getReadonlyContentScheme(): string;
```
Also, it should be put right next to `openReadOnlyContent` in `index.d.ts`